zyte_parsers/gtin.py

```diff
@@ -0,0 +1,161 @@
+import re
+from typing import Optional, Union
+
+import attr
+from gtin.validator import is_valid_GTIN
+from stdnum import isbn, ismn, issn
+
+from . import SelectorOrElement
+from .api import input_to_element
+from .utils import extract_text
+
+
+@attr.s(frozen=True, auto_attribs=True)
+class Gtin:
+    type: str
+    value: str
+
+
+GTIN_MATCH_SPECIAL_CHARACTER_REGEX = re.compile(r"[^0-9a-zA-Z]")
+GTIN_MATCH_NON_NUMERIC_REGEX = re.compile(r"[^0-9]")
+
+# Consider only those prefix which have some numeric values as these
+# values interfere with the gtin id extraction
+GTIN_PREFIX = [
+    "isbn13",
+    "isbn10",
+    "ean13",
+    "gtin8",
+    "gtin12",
+    "gtin13",
+    "gtin14",
+]
+GTIN_PREFIX_REGEX = re.compile("|".join(GTIN_PREFIX), re.IGNORECASE)
+GTIN_CENTER_REGEX = re.compile(r"^\D*|\D*$")
+
+
+def extract_gtin(node: Union[SelectorOrElement, str]) -> Optional[Gtin]:
+    """Extract a GTIN (Global Trade Item Number) from a node that contains its text.
```

Comment:
It seems that unlike other extactors, here we also support text as an input, but it's not documented.

Options:

* document it
* add support for text to other extractors when it makes sense, for consistency
* remove text support, for consistency
* something else?

What do you think?

Reply:
We should document it and add support to other extractors as recently asked in https://github.com/zytedata/zyte-common-items/pull/69#pullrequestreview-1790255775